packages/datatrak-web/src/api/mutations/useTupaiaRedirect.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useMutation } from 'react-query';
+import { useCurrentUser } from '../CurrentUserContext';
+import { post } from '../api';
+
+const TUPAIA_REDIRECT_URL = process.env.REACT_APP_TUPAIA_REDIRECT_URL || 'https://tupaia.org'
+
+export const useTupaiaRedirect = () => {
+  const user = useCurrentUser();
+
+  return useMutation<any, Error, Record<never, any>, unknown>(
+    () => {
+      return post('generateLoginToken');
+    },
+    {
+      onSuccess: ({ token }) => {
+        window.open(`${TUPAIA_REDIRECT_URL}/${user.project?.code}/${user.project?.code}?loginToken=${token}`, '_blank');
```

Comment:
Quick question - should we be putting the country code in here instead of the project code twice?

Reply:
I don't think the link is available within a country context. It's just on your main datatrak landing page which is only in a project scope right?

project/project links you to the project entity which is the main landing page for all projects

Reply:
Done 👍